src/Api/Serializer/AbstractSerializer.php

```diff
@@ -83,6 +94,18 @@ public function getAttributes($model, array $fields = null)
 
         $attributes = $this->getDefaultAttributes($model);
 
+        foreach (array_reverse(array_merge([static::class], class_parents($this))) as $class) {
+            if (isset(static::$mutators[$class])) {
+                foreach (static::$mutators[$class] as $callback) {
+                    $attributes = array_merge(
+                        $attributes,
+                        $callback($attributes, $model, $this)
+                    );
```

Comment:
would we prefer simply calling the callback instead of doing an array_merge here ? since extension devs can choose to take the $attributes parameter by reference.

Reply:
From a functional programming perspective, I prefer return and merge over mutation